src/isomorphic/devtools/ReactComponentTreeDevtool.js

```diff
@@ -19,6 +19,7 @@ var rootIDs = [];
 function updateTree(id, update) {
   if (!tree[id]) {
     tree[id] = {
+      source: null,
```

Comment:
Yes, but AFAIK we plan to kill that code and make React DevTools use this tool instead. (Or implement a similar one using the same events.) For older versions of React, we will add a shim to React DevTools that emits these events so that its core code does not rely on traversing the tree. This way it would become compatible with incremental reconciler.